### PR TITLE
TIG-3247 Add schedule_global_tasks dry-run to genny

### PR DESCRIPTION
**Pull request by vrachev**
Ran the test on https://github.com/mongodb/genny/commit/07eb608c48cd15df042648d8b34d61587d373b79 and it correctly failed and printed out the syntax error.

### CODE REVIEW

File: src/lamplib/src/tests/test_auto_tasks.py [hunk 1]
```diff
@@ -430,5 +434,38 @@ def test_patch_tasks(self):
         )
 
 
+def test_generate_all_tasks():
```
Comment (aggrand) [inline src/lamplib/src/tests/test_auto_tasks.py:437]: Could we rename this to `test_dry_run_all_tasks` or so? There's already a `test_all_tasks` further up that might be confusing.

File: src/lamplib/src/tests/test_auto_tasks.py [hunk 2]
```diff
@@ -430,5 +434,38 @@ def test_patch_tasks(self):
...
+    """This is a dry run of schedule_global_auto_tasks."""
+    genny_repo_root = os.environ.get("GENNY_REPO_ROOT", None)
+    workspace_root = "."
+    if not genny_repo_root:
+        raise Exception(
+            f"GENNY_REPO_ROOT env var {genny_repo_root} either not set or does not exist. "
+            f"This is set when you run through the 'run-genny' wrapper"
+        )
+    try:
+        reader = YamlReader()
```
Comment (aggrand) [inline src/lamplib/src/tests/test_auto_tasks.py:447]: Running locally I get `Exception: File ./expansions.yml not found.`. Are we able to mock out the expansion data? Or should we put an expansion file fixture somewhere?

**Review by aggrand — CHANGES_REQUESTED** [review]
Wow thanks for getting this up so fast! Will be super useful. Just had a couple nitpicks.

File: src/lamplib/src/tests/test_auto_tasks.py (hunk 2 above)
Comment (vrachev) [reply src/lamplib/src/tests/test_auto_tasks.py:447]: Ah shoot yeah I thought I was mocking it out already

**Comment by vrachev** [comment]
@aggrand RFAL

**Review by aggrand — APPROVED** [review]
LGTM, thanks a bunch for this!

**Comment by vrachev** [comment]
evergreen merge

**Comment by vrachev** [comment]
evergreen merge

**Pull request merged**